shopfloor/models/stock_move_line.py

```diff
@@ -21,3 +21,86 @@ class StockMoveLine(models.Model):
 
     # allow domain on picking_id.xxx without too much perf penalty
     picking_id = fields.Many2one(auto_join=True)
+
+    def _split_pickings_from_source_location(self):
+        """Ensure that the related pickings will have the same source location.
+
+        Some pickings related could have other unrelated move lines, as such we
+        have to split them to contain only the move lines related to the expected
+        source location.
+
+        Example:
+
+            Initial data:
+
+                PICK1:
+                    - move line with source location LOC1
+                    - move line with source location LOC2
+                PICK2:
+                    - move line with source location LOC2
+                    - move line with source location LOC3
+
+            Then we process move lines related to LOC2 with this method, we get:
+
+                PICK1:
+                    - move line with source location LOC1
+                PICK2:
+                    - move line with source location LOC3
+                PICK3:
+                    - move line with source location LOC2
+                    - move line with source location LOC2
+
+        Return the new picking (in case a split has been made), or the current
+        related pickings.
+        """
+        location_src_to_process = self.location_id
+        if location_src_to_process:
+            assert (
+                len(location_src_to_process) == 1
+            ), "Move lines processed have to share the same source location."
```

Comment:
If it's an assert, it means we are never ever supposed to fall here and that's supposed to be for a dev.
If we can reach this, then we should need translation and UserError instead of an assert.

Reply:
We shouldn't reach this part from the location content transfer scenario, but if we want to use this method from another context, then yes a `UserError` is what we want, I'll update it that way.

Reply:
Updated